### ISSUE TITLE
Difference in behavior of env case: env PATH=/bin OLDPWD=/root cd /

### ISSUE THREAD

**Issue by Ant0wan** [Ant0wan/42sh#10]
Our shell launch cd when bash seems not to find binary "cd". Return value is also different
```
~/42sh$ env PATH=/bin OLDPWD=/root cd /
~/42sh$ echo $?
0
```
bash --posix
```
bash-5.0$ env PATH=/bin OLDPWD=/root cd /
env: ‘cd’: No such file or directory
bash-5.0$ echo $?
127
```

**Comment by Ant0wan** [comment]
related issue
```~/42sh$ PATH=/new env -i env
PATH=/new
~/42sh$ echo $?
0
```
bash --posix output
```bash-5.0$ PATH=/new env -i env
bash: env: command not found
bash-5.0$ echo $?
127
```

**Comment by Taiwing** [comment]
The first one is not a problem. It works because env is a builtin for us, whereas bash calls /usr/bin/env and asks it to find 'cd' on the PATH (which of course it doesnt find). Our env builtin is able to launch builtins. We can deactivate it, but I don't really see the use.

The second one is also not a problem for the same reason. However it shouldn't show the PATH variable. This is a specific case when env is called with env, it ignores the '-i' option. Maybe deactivating builtin execution is the easiest and safest choice ?

**Comment by Ant0wan** [comment]
When it comes to execute job control builtins, the shell/terminal is stuck.
`env fg` puts in foreground background jobs, then the terminal is messed up.

**Comment by Taiwing** [comment]
Fixed in commit e3d3892
